bundles/org.openhab.core.io.net/src/main/java/org/openhab/core/io/net/exec/ExecUtil.java

```diff
@@ -104,22 +105,33 @@ public static void executeCommandLine(String... commandLine) {
             } else if (process.waitFor(timeout.toMillis(), TimeUnit.MILLISECONDS)) {
                 exitCode = process.exitValue();
             } else {
-                logger.warn("Timeout occurred when executing commandLine '{}'", commandLine);
+                logger.warn("Timeout occurred when executing commandLine '{}'", Arrays.toString(commandLine));
                 break cleanup;
             }
             if (exitCode == 0) {
                 return outputFuture.get();
             } else {
-                logger.debug("exit code '{}', result '{}', errors '{}'", exitCode, outputFuture.get(),
-                        errorFuture.get());
+                if (logger.isDebugEnabled()) {
+                    logger.debug("exit code '{}', result '{}', errors '{}'", exitCode, outputFuture.get(),
+                            errorFuture.get());
+                }
                 return null;
             }
         } catch (ExecutionException e) {
-            logger.warn("Error occurred when executing commandLine '{}'", commandLine, e.getCause());
+            if (logger.isDebugEnabled()) {
+                logger.debug("Error occurred when executing commandLine '{}'", Arrays.toString(commandLine),
```

Comment:
I would keep logging this as WARN level here as the severity isn't reduced, just because debug logging is enabled. All that is added is that the exception will be logged in that case.
```suggestion
                logger.warn("Error occurred when executing commandLine '{}'", Arrays.toString(commandLine),
```

Reply:
done